lib/sender.js

```diff
@@ -186,13 +184,31 @@ FluentSender.prototype._connect = function(callback){
 
 FluentSender.prototype._flushSendQueue = function() {
   var self = this;
-  var pos = self._sendQueue.length - self._sendQueueTail - 1;
-  var item = self._sendQueue[pos];
+  if (self._flushingSendQueue)
+    return;
+
+  self._flushingSendQueue = true;
+  process.nextTick(function waitToWrite() {
```

Comment:
`waitToWrite` should probably be declared outside of `_flushSendQueue`, otherwise you're declaring that function with every chunk sent